src/stream/src/common/table/state_table.rs

```diff
@@ -985,12 +1027,12 @@ impl<S: StateStore, W: WatermarkBufferStrategy> StateTable<S, W> {
         pk_range: &(Bound<impl Row>, Bound<impl Row>),
         // Optional vnode that returns an iterator only over the given range under that vnode.
         // For now, we require this parameter, and will panic. In the future, when `None`, we can
-        // iterate over each vnode that the `StateTable` owns.
+        // iterate over each vnode that the `StateTableInner` owns.
         vnode: VirtualNode,
-    ) -> StreamExecutorResult<RowStreamWithPk<'_, S>> {
+    ) -> StreamExecutorResult<RowStreamWithPk<'_, S, SD>> {
         Ok(deserialize_row_stream(
             self.iter_with_pk_range_inner(pk_range, vnode).await?,
-            self.row_deserializer.clone(),
+            self.row_serde.clone(),
```

Comment:
Do you mean the `clone` here? Actually I think this is unnecessary since the returned `Stream` also captures the lifetime of borrowed `self` (see `'_`). Could you please help to try passing a reference here?

If you find the compiler hard to deal with, we can also use `Arc` here.

Reply:
done 😇